analyzer/src/main/kotlin/managers/Pub.kt

```diff
@@ -241,20 +243,72 @@ class Pub(
             log.info { "Successfully read lockfile." }
 
             val parsePackagesResult = parseInstalledPackages(lockFile, labels)
+            val dependenciesScope = parseScope("dependencies", manifest, lockFile, parsePackagesResult.packages, labels)
+
+            if (containsFlutterSdk(workingDir)) {
+                // Run Gradle dependency resolution. This is required because for Flutter projects, Pub needs to run
+                // first to generate the local.properties file by using `flutter pub get`.
+                val gradleFactory = Gradle.Factory()
+                val gradleDefinitionFiles =
+                    findManagedFiles(analysisRoot, listOf(gradleFactory)).getOrDefault(gradleFactory, emptyList())
+
+                if (gradleDefinitionFiles.isNotEmpty()) {
+                    println("Found ${gradleDefinitionFiles.size} ${gradleFactory.managerName} project(s) at:")
+
+                    gradleDefinitionFiles.forEach {
+                        val relativePath = it.toRelativeString(workingDir).takeIf { it.isNotEmpty() } ?: "."
+
+                        println("\t$relativePath")
+                    }
+                }
+
+                log.info { "Running Gradle analysis for Flutter project." }
+
+                // TODO: This ignores the AnalyzerCommand's `--package-managers` option and always runs Gradle for
+                //  Flutter projects.
+                val gradleAnalyzerResult = Gradle(
```

Comment:
It's a bit unfortunate that we're using both the Gradle factory (see above) and a custom Gradle instance here. Can somehow align on using only either one?